include/HubFramework/HUBComponentModelBuilder.h

```diff
@@ -26,10 +26,30 @@
 
 @protocol HUBComponentImageDataBuilder;
 @protocol HUBComponentTargetBuilder;
+@protocol HUBComponentModelBuilder;
 
 NS_ASSUME_NONNULL_BEGIN
 
 /**
+ *  Protocol defining the public delegate API for a builder that builds component model objects
```

Comment:
The delegate should not be part of the public API, as it's only needed internally. So move this to `HUBComponentModelBuilderImplementation.h`.